### PR TITLE
feat: cleaning up the synchronizer

### PR DESCRIPTION
**Pull request by Freyskeyd**
# Description

This PR aims to add and update passive checkpoint sync implementation and add tests on the storage part.

## PR Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: crates/topos-tce-api/src/grpc/mod.rs [hunk 1]
```diff
@@ -213,58 +215,38 @@ impl ApiService for TceGrpcService {
...
-                    "Can't get last pending certificates: {e}"
-                ))),
+        let last_pending_certificate = self
+            .store
+            .get_pending_certificates_for_subnets(&subnet_ids)
+            .map_err(|e| Status::internal(format!("Can't get last pending certificates: {e}")))?
+            .into_iter()
+            .map(|(subnet_id, (index, maybe_certificate))| {
+                (
+                    base64::engine::general_purpose::STANDARD.encode(subnet_id.as_array()),
```
Comment (dvdplm) [inline crates/topos-tce-api/src/grpc/mod.rs:231]: So this line looked annoyingly long and I went to check the docs to see if there's not something less fugly, but no, this is how it's supposed to be used (!). I did some more digging and it seems like the `base64` maintainer is struggling to keep up maintenance. One notable feature that is missing is constant time encoding/decoding. 

What do you think about switching to the RustCrypto `base64ct` crate instead? I don't think being constant time matters in this particular code, but if we have a single crate in the code base that happens to be constant time then we save our future selves a papercut: no need to even think about "so do I need the constant time base64 crate here or is the other one ok?".

File: crates/topos-tce-api/src/runtime/builder.rs [hunk 2]
```diff
@@ -90,6 +90,12 @@ impl RuntimeBuilder {
         let (api_event_sender, api_event_receiver) = mpsc::channel(2048);
 
         let (health_reporter, tce_status, grpc) = ServerBuilder::default()
+            .with_store(
+                self.store
+                    .clone()
+                    .take()
+                    .expect("Unable to build gRPC Server, Store is missing"),
```
Comment (dvdplm) [inline crates/topos-tce-api/src/runtime/builder.rs:97]: Same comment as above.

**Review by dvdplm — APPROVED** [review]

File: crates/topos-tce-api/src/grpc/builder.rs [hunk 3]
```diff
@@ -59,7 +67,15 @@ impl ServerBuilder {
             status: status.clone(),
         });
 
-        let service = ApiServiceServer::new(TceGrpcService { command_sender });
+        let store = self
+            .store
+            .take()
+            .expect("Cannot build GraphQL server without a FullNode store");
```
[the remark this answers is not in the document]
Comment (Freyskeyd) [reply crates/topos-tce-api/src/grpc/builder.rs:73]: That's part of the refactor effort that I initiate, but it's not the scope of the PR to remove builders.

File: crates/topos-tce-api/src/grpc/mod.rs (hunk 1 above)
Comment (Freyskeyd) [reply crates/topos-tce-api/src/grpc/mod.rs:231]: Handled by another PR

**Pull request merged**